### PR TITLE
Use prebuilt rustc binary

### PR DESCRIPTION
**Pull request by f-a-a**
This PR is a (second) attempt to solve #386.

This approach proposes to include **rustc** and **cargo** binaries in the prebuilt folder.

### CODE REVIEW

**Comment by hayd** [comment]
Can we upload the latest rust stable (1.31.1)?
(That would fix/enable #1306.)

File: .travis.yml [hunk 1]
```diff
@@ -28,29 +28,12 @@ install:
...
-- |-
-  # Install Rust.
-  # TODO(ry) Include rustc in third_party.
-  # https://github.com/denoland/deno/issues/386
-  if [ ! $(rustc --version | grep 1.30.0) ]; then
-    curl -sSf https://sh.rustup.rs | sh -s -- -y --default-toolchain 1.30.0
-    rustup default 1.30.0
-  fi
-  rustc --version
-  cargo --version
```
Comment (ry) [inline .travis.yml]: enticing!

**Comment by f-a-a** [comment]
@ry the initial approach was bogus xD I've updated the PR now that seems to be working with ~/.rustup and ~/.cargo removed.

Currently the `./tools/setup.py` step passes but I'm getting errors in `./tools/run_format.py` and can't pin point what exactly because it's running OK locally.

**Comment by f-a-a** [comment]
@ry I'm not sure if I should be bundling rust as `third_party` dependency here, let me know what you think.

**Pull request closed without merging**